### PR TITLE
Introduction of gcc-11 to CI

### PR DESCRIPTION
**Pull request by pwisniewskimobica**
Switch to gcc11 and g++11

### CODE REVIEW

**Comment by pwisniewskimobica** [comment]
We want to introduce compilation with gcc-11 in Flang CI. 
This PR lets us build the gcc-11 artifacts for later use in Flang repository CI which is done in 
https://github.com/flang-compiler/flang/pull/1190

Kind request for your reviews, @bryanpkc , @shivaramaarao , @mleair , @kiranchandramohan
Thank you in advance!

File: .github/workflows/pre-compile_llvm.yml [hunk 1]
```diff
@@ -35,18 +32,19 @@ jobs:
         run: |
           sudo add-apt-repository ppa:ubuntu-toolchain-r/ppa
```
Comment (bryanpkc) [inline .github/workflows/pre-compile_llvm.yml:32]: I think the logic would be easier to understand if you add the following here, and delete `env:` section above:
```
echo "cpp=g++" >> $GITHUB_ENV
```

**Comment by kiranchandramohan** [comment]
I think release 10 is not used now. So you may close this.

File: .github/workflows/pre-compile_llvm.yml (hunk 1 above)
Comment (pwisniewskimobica) [reply .github/workflows/pre-compile_llvm.yml:32]: fixed

**Comment by kiranchandramohan** [comment]
Closing since we do not support release_10 now.

**Pull request closed without merging**